src/autoscaler/scalingengine/cmd/scalingengine/main.go

```diff
@@ -98,19 +99,53 @@ func main() {
 	synchronizer := schedule.NewActiveScheduleSychronizer(logger.Session("synchronizer"), schedulerDB, scalingEngineDB, scalingEngine,
 		conf.Synchronizer.ActiveScheduleSyncInterval, eClock)
 
-	members := grouper.Members{
+	nonLockMembers := grouper.Members{
 		{"http_server", httpServer},
+	}
+
+	lockMembers := grouper.Members{
 		{"schedule_synchronizer", synchronizer},
 	}
 
-	monitor := ifrit.Invoke(sigmon.New(grouper.NewOrdered(os.Interrupt, members)))
+	if conf.EnableDBLock {
+		const lockTableName = "scalingengine_lock"
+		guid, err := helpers.GenerateGUID(logger)
+		if err != nil {
+			logger.Error("failed-to-generate-guid", err)
+		}
+		logger.Debug("database-lock-feature-enabled")
+		var lockDB db.LockDB
+		lockDB, err = sqldb.NewLockSQLDB(conf.DBLock.LockDB, lockTableName, logger.Session("lock-db"))
+		if err != nil {
+			logger.Error("failed-to-connect-lock-database", err, lager.Data{"dbConfig": conf.DBLock.LockDB})
+			os.Exit(1)
+		}
+		defer func() {
+			lockDB.Release(guid)
+			lockDB.Close()
+		}()
+
+		sedl := sync.NewDatabaseLock(logger)
+		dbLockMaintainer := sedl.InitDBLockRunner(conf.DBLock.LockRetryInterval, conf.DBLock.LockTTL, guid, lockDB)
+		lockMembers = append(grouper.Members{{"db-lock-maintainer", dbLockMaintainer}}, lockMembers...)
+	}
 
+	nonLockMonitor := ifrit.Invoke(sigmon.New(grouper.NewOrdered(os.Interrupt, nonLockMembers)))
 	logger.Info("started")
 
-	err = <-monitor.Wait()
-	if err != nil {
-		logger.Error("exited-with-failure", err)
-		os.Exit(1)
+	lockMonitor := ifrit.Invoke(sigmon.New(grouper.NewOrdered(os.Interrupt, lockMembers)))
```

Comment:
there is still issue  for the instance that does not have the lock to run synchronizer.  It will hold in line 136,  and will not be able to monitor the errors happened in non-locked processes.